dlib/dnn/cpu_dlib.cpp

```diff
@@ -1699,6 +1699,28 @@ namespace dlib
             }
         }
 
+        void tensor_conv::forward_without_setup(
```

Comment:
Ok.  I'll look into it when you eventually want to merge.  

But I should point out now that I won't merge pull requests that have incorrect/missing comments/documentation.  I have some instructions about this here: http://dlib.net/compile.html.  I'm a pretty big fan of precise interface specifications :)